packages/flame/lib/src/components/clip_component.dart

```diff
@@ -0,0 +1,132 @@
+import 'dart:ui';
+
+import 'package:flame/components.dart';
+import 'package:flame/experimental.dart';
+
+/// {@template clip_component}
+/// A component that will clip its content.
+/// {@endtemplate}
+abstract class ClipComponent extends PositionComponent {
+  /// {@macro clip_component}
+  ///
+  /// Clips the canvas based its shape and size.
+  ClipComponent({
+    super.position,
+    super.size,
+    super.scale,
+    super.angle,
+    super.anchor,
+    super.children,
+    super.priority,
+  });
+
+  late Path _path;
+  late Shape _shape;
+
+  @override
+  Future<void> onLoad() async {
+    _prepare();
+    size.addListener(_prepare);
```

Comment:
I need to look into how addListener on Size works. Does it need to be disposed?

Reply:
No needed, this is subscribing to its own size, and it should live until the component live, so when the component is removed and garbage collected, this will go along

Reply:
I see very clever and simple implementation. Love it!